syntax = "proto3";

package registry;

service RegistryService {
  rpc RegisterService (RegistryRequest) returns (RegistryResponse) {}
}

message RegistryRequest {
  string serviceName = 1;
  string serviceAddress = 2;
  int32 servicePort = 3;
}

message Status {
  int32 code = 1;
  string message = 2;
}

message RegistryResponse {
  string serviceName = 1;
  Status status = 2;
}
